sycl/include/sycl/ext/oneapi/reduction.hpp

```diff
@@ -876,13 +1366,16 @@ template <typename KernelName, typename KernelType, int Dims, class Reduction>
 std::enable_if_t<!Reduction::has_fast_atomics && Reduction::has_fast_reduce>
 reduCGFuncImpl(handler &CGH, KernelType KernelFunc, const range<Dims> &Range,
                const nd_range<1> &NDRange, Reduction &Redu) {
+  constexpr size_t NElements = Reduction::num_elements;
   size_t WGSize = NDRange.get_local_range().size();
   size_t NWorkGroups = NDRange.get_group_range().size();
 
   bool IsUpdateOfUserVar = !Reduction::is_usm && !Redu.initializeToIdentity();
-  auto PartialSums = Redu.getWriteAccForPartialReds(NWorkGroups, CGH);
-  auto Out =
-      (NWorkGroups == 1) ? PartialSums : Redu.getWriteAccForPartialReds(1, CGH);
+  auto PartialSums =
+      Redu.getWriteAccForPartialReds(NWorkGroups * NElements, CGH);
```

Comment:
Ough, I start forgetting how things worked in reduction implementation.
This place looks a bit concerning. I remember that NWorkGroups=1 is this call handled some special way and with sycl::span the value =1 is never passed to getWriteAccForPartialReds.
John, did you try running read-write accessor + sycl::span?